server/gitlab/gitlab.go

```diff
@@ -47,23 +48,34 @@ type gitlab struct {
 	checkGroup        func(projectNameWithGroup string) error
 }
 
+// Scope identifies the scope of a webhook
+type Scope int
+
+const (
+	//Group is a type for group hooks
+	Group Scope = iota
+	//Project is a type for project hooks
+	Project
+)
+
+func (s Scope) String() string {
+	return [...]string{"group", "project"}[s]
+}
+
 // New return a client to call GitLab API
 func New(enterpriseBaseURL string, gitlabGroup string, checkGroup func(projectNameWithGroup string) error) Gitlab {
 	return &gitlab{enterpriseBaseURL: enterpriseBaseURL, gitlabGroup: gitlabGroup, checkGroup: checkGroup}
 }
 
 func (g *gitlab) gitlabConnect(token oauth2.Token) (*internGitlab.Client, error) {
-	ctx := context.Background()
-	ts := oauth2.StaticTokenSource(&token)
-	tc := oauth2.NewClient(ctx, ts)
-
 	if len(g.enterpriseBaseURL) == 0 {
-		return internGitlab.NewOAuthClient(tc, token.AccessToken), nil
+		return internGitlab.NewOAuthClient(token.AccessToken)
 	}
 
-	client := internGitlab.NewOAuthClient(tc, token.AccessToken)
-	if err := client.SetBaseURL(g.enterpriseBaseURL); err != nil {
-		return nil, fmt.Errorf("can't set base url to GitLab client lib: %w", err)
+	client, err := internGitlab.NewOAuthClient(token.AccessToken)
```

Comment:
Not setting base url will make requests go to the default `gitlab.com` URL.
What was the reason for changing `gitlabConnect` method?

Reply:
You are absolutely right. The reason for the modifications to ```gitlabConnect``` was because of changes in go-gitlab. The types of arguments for ```NewOAuthClient``` changed, and ```SetBaseURL``` became a private function. I must have changed it to get this feature to compile and the forgot that the ```gitlabConnect``` method needs a re-write.

Reply:
I just pushed a change and tested it with Gitlab.com. Here is the link to the breaking change commit that I updated gitlabConnect to address. https://github.com/xanzy/go-gitlab/issues/826